### ISSUE TITLE
Should convnext_proj layer have disabled gradient computation?

### ISSUE THREAD

**Issue by JanKrcek** [shilinyan99/AIDE#19]
Greetings and big thanks for your contribution to the AI-gen detection research.

The idea of using semantic information in addition to the visual noise for classification is very cool. 

Could you please clarify something to me?
Let's consider the case of training the whole AIDE network **from scratch.**

After reading the paper, I was under impression that according to architecture diagram in Figure 2., the convnext Projection Layer is one of the layers being trained. Which made perfect sense, however when I looked into the code [here](https://github.com/shilinyan99/AIDE/blob/main/models/AIDE.py#L280), that does not seem to be the case.
As it sits indented within the `torch.no_grad():` block, the weights will stay unchanged in their random initialization state forever.

Please do correct me if I am wrong, but It looks like that might not be behaving as intended.

**Comment by yangjiazhne** [comment]
I have the same concern. When I tried to reproduce the ablation study, I found that after removing the **PATCHWISE FEATURE EXTRACTION** branch, the CLIP branch couldn’t propagate gradients (and it seems that even within the full framework the CLIP branch can’t backpropagate). After moving the MLP branch out of `torch.no_grad()` and training on the full GenImage SD v1.4 training set, the accuracy on the VQDM test set was below 60%.
